File: 151002/Redkovskiy/StoryTeller/src/main/java/com/example/storyteller/model/Story.java
package com.example.storyteller.model;

import jakarta.persistence.*;
import lombok.Data;

import java.sql.Date;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Data
@Entity
@Table(name = "tbl_story")
public class Story {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(length = 64)
    private String title;

    @Column(length = 2048)
    private String content;

    private LocalDateTime created;

    private LocalDateTime modified;

    @ManyToOne
    @JoinColumn(name = "creator_id", referencedColumnName = "id")
    private Creator creator;

    @OneToMany(mappedBy = "story", cascade = CascadeType.ALL)
    Set<Post> posts = new HashSet<>();

    @OneToMany(mappedBy = "story")
    private Set<StoryMarker> markers = new HashSet<>();

}

File: 151004/Akimenko/DC_task_rest/WebApplicationDC1/WebApplicationDC1/Entities/DataModels/Story.cs
using System.ComponentModel.DataAnnotations;

namespace WebApplicationDC1.Entity.DataModel
{
    public class Story(string title, string content, DateTime created, DateTime modified) : Entity
    {
        [MinLength(2)]
        public string Title { get; set; } = title;
        [MinLength(4)]
        public string Content { get; set; } = content;
        public DateTime Created { get; set; } = created;
        public DateTime Modified { get; set; } = modified;
        public ICollection<Post> Posts { get; set; } = [];
        public ICollection<Sticker> Stickers { get; set; } = [];
        public Creator Creator { get; set; } = null!;

        public Story() : this(string.Empty, string.Empty, DateTime.Now, DateTime.Now) { }
        public Story(int id, Creator creator, string title, string content, DateTime created, DateTime modified)
            : this(title, content, created, modified)
        {
            Id = id;
            Creator = creator;
        }
    }
}
